File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        if (s.length()==0) return 0;
        //sliding window
        int left =0;
        int right = 0;
        int longest =1;
        //makes a map that updates for each window
        HashSet<Character> map = new HashSet<>();
        

        while(right<s.length()){
            //gets the character at the right of the window
            char ch = s.charAt(right);
            while(map.contains(ch)){
                map.remove(s.charAt(left));
                left++;
            }
            map.add(ch);
            longest = Math.max(longest, right-left+1);
            right++;
                
         }
            
        return longest;
    }
}

File: Data Structures & Algorithms/remove-element/submission-1.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        right = len(nums)-1
        left = 0

        while left <= right:
            if nums[left] ==val:
               
                nums[left] = nums[right]
                nums[right] = val
                right-=1
            else:
                left+=1

        return left